### PR TITLE
TEAM CYMRU - IP to ASN

### PR DESCRIPTION
**Pull request by michal-dagan**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/CIAC-3401)

## Description
Team Cymru IP to asn service allows us to send IOCs in bulk (up to 10k)

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+
+
+def main() -> None:
+    """
+    main function, parses params and runs command functions
+    """
+
+    # if your Client class inherits from BaseClient, SSL verification is
+    # handled out of the box by it, just pass ``verify_certificate`` to
+    # the Client constructor
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:243]: Please remove the comment

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    verify_certificate = not demisto.params().get('insecure', False)
+
+    # if your Client class inherits from BaseClient, system proxy is handled
+    # out of the box by it, just pass ``proxy`` to the Client constructor
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:247]: Please remove the comment

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+    proxy = demisto.params().get('proxy', False)
+
+    demisto.debug(f'Command being called is {demisto.command()}')
+    try:
+        client = Client()
+
+        if demisto.command() == 'test-module':
+            # This is the call made when pressing the integration Test button.
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:255]: Please remove the comment

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+    :param Client: client to use
+
+    :type args: ``Dict[str, Any]``
+    :param args: All command arguments - 'entry_id', 'delimiter'
+    :return: CommandResults object containing the results of the lookup action as returned from the API
+    and its readable output.
+    """
+
+    if args.get('entry_id'):
+        demisto.debug('getting the path of the file from its entry_id')
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:222]:
```suggestion
        demisto.debug('Using the entry_id to find the file's path')
```

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+        get_file_path_res = demisto.getFilePath(args.get('entry_id'))
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:223]:
```suggestion
        file_path = demisto.getFilePath(args.get('entry_id'))
```

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+
+def parse_file(get_file_path_res: dict[str, str], delimiter: str = ",") -> List[str]:
+    """
+    Parses the given file line by line to list.
+    :param delimiter: delimiter by which the content of the list is seperated.
+    :param get_file_path_res: Object contains file ID, path and name
+    :return: bulk list of the elements in the file
+    """
+    bulk_list = []
+    with open(get_file_path_res['path']) as file:
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:121]:
```suggestion
    with open(get_file_path_res.get('path')) as file:
```

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+    :param ip: string to add in the dummy dict that is returned
+
+    :return: dict containing the result of the lookup action as returned from the API (asn, cc, owner, etc.)
+    """
+    raw_result = client.lookup(ip)
+    return vars(raw_result) if raw_result else None
+
+
+def team_cymru_bulk_whois(client: Client, bulk: List[str]) -> Optional[Dict[str, Any]]:
+    """Perform lookups by bulk of ip addresses, returning a dictionary of ip -> record (ASN, Country Code, and Netblock Owner.)
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:37]: Sounds better?
```suggestion
    """Performs bulk IP address searches, returning a dictionary of IP records (ASN, Country Code, and Netblock Owner.)
```

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+
+    :type client: ``Client``
+    :param client: cymruwhois client to use
+    :type bulk: ``list``
+    :param bulk: list of ip addresses
+
+    :return: dict containing the result of the lookupmany action as returned from the API (asn, cc, owner, etc.)
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:44]:
```suggestion
    :return: Dictionary contains the results of the lookupmany API call.
```

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+        readable_output=human_readable,
+        raw_response=ip_data,
+        outputs_prefix='TeamCymru.IP',
+        outputs_key_field=outputs_key_field,
+        indicator=indicator,
+        outputs=entry_context
+    )
+
+
+def validate_ip_addresses(ips_list: list[str]):
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:95]: Please add typing for the return value

File: Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py [hunk 10]
```diff
@@ -0,0 +1,274 @@
...
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        result = team_cymru_ip(client, '8.8.8.8')
+        # This  should validate all the inputs given in the integration configuration panel,
+        # either manually or by using an API that uses them.
```
Comment (bziser) [inline Packs/TeamCymru/Integrations/TeamCymru/TeamCymru.py:171]: remove

File: Packs/TeamCymru/pack_metadata.json [hunk 11]
```diff
@@ -0,0 +1,19 @@
+{
+    "name": "Team Cymru",
+    "description": "## Team Cymru is an internet security firm that offers research services making the internet a more secure place. ##",
```
Comment (bziser) [inline Packs/TeamCymru/pack_metadata.json:3]:
```suggestion
    "description": "Team Cymru is an internet security firm that offers research services making the internet a more secure place.",
```

**Review by bziser — COMMENTED** [review]
Great work! 🤗

**Comment by ShirleyDenkberg** [comment]
@bziser Doc review completed.

**Review by bziser — APPROVED** [review]

**Pull request merged**